dart/lib/src/protocol/sentry_event.dart

```diff
@@ -262,6 +262,8 @@ class SentryEvent {
       json['exception'] = {
         'values': [exception.toJson()].toList(growable: false)
       };
+    } else if (stackTrace is SentryStackTrace) {
+      json['stacktrace'] = (stackTrace as SentryStackTrace).toJson();
```

Comment:
why this change? an exception has a stacktrace field already

Reply:
it's for case we want attach a stacktrace without exception ( for captureMessage and captureEvent )

Reply:
guess this would be necessary for captureMessage() as theres no exception

Reply:
ooops I just read that stacktrace should never be a top level property

Reply:
maybe we add one more field to the SentryEvent, because the docs about the stackTrace:

`Can be `null`, a [String], or a [StackTrace].`

it'd be confusing to add one more type, same as we have `throwable` and `exception`
wdyt?

Reply:
other solution would be to add the threads list

Reply:
what about a `event.attachedStackTrace` property ? cause a `threads` interface only for a single property could also be confusing ?

Reply:
this is just a toJson method, it should not have any business logic here

Reply:
I meant adding a `attachedStackTrace ` field to SentryEvent
 
```dart 
if (exception != null) {
      json['exception'] = {
        'values': [exception.toJson()].toList(growable: false)
      };
} else if (attachedStackTrace != null) {
      json['threads'] = {
        'values': [
          {
            'id': 0,
            'stacktrace': (stackTrace as SentryStackTrace).toJson(),
          }
        ]
      };
    }
```
WDYT?

Reply:
we dont need `attachedStackTrace` to know if we should make the `threads` list or not, this could be done by checking if the exception is null or not.
by big concern here is `stackTrace as SentryStackTrace` as mentioned before.

Reply:
oh yes sorry I meant 
```dart
if (exception != null) {
      json['exception'] = {
        'values': [exception.toJson()].toList(growable: false)
      };
} else if (attachedStackTrace != null) {
      json['threads'] = {
        'values': [
          {
            'id': 0,
            'stacktrace': attachedStackTrace.toJson(),
          }
        ]
      };
    }
```

with `attachedStackTrace` being a SentryStackTrace.
Do you prefer the threads property options ?

Reply:
would be possible to make `stackTrace` strong typed as `SentryStackTrace`? and all the captureEvent, captureException etc takes the given stackTrace (from a try catch) as a melhod param, so we can pass the stackTrace along and we dont really need the stackTrace on the event being dynamic.

Reply:
so then your implementation would work and we would not need to call it `attachedStackTrace` which is confusing